services/elasticache/pkg/resource/replication_group/custom_update_api.go

```diff
@@ -0,0 +1,388 @@
+	r *resource,
+) (*svcsdk.IncreaseReplicaCountInput, error) {
+	res := &svcsdk.IncreaseReplicaCountInput{}
+
+	res.SetApplyImmediately(true)
+	if r.ko.Spec.ReplicationGroupID != nil {
+		res.SetReplicationGroupId(*r.ko.Spec.ReplicationGroupID)
+	}
+	if r.ko.Spec.NumNodeGroups != nil {
+		res.SetNewReplicaCount(*r.ko.Spec.ReplicasPerNodeGroup)
+	}
+
+	if r.ko.Spec.NodeGroupConfiguration != nil {
+		f13 := []*svcsdk.ConfigureShard{}
+		for _, f13iter := range r.ko.Spec.NodeGroupConfiguration {
+			f13elem := &svcsdk.ConfigureShard{}
+			if f13iter.NodeGroupID != nil {
+				f13elem.SetNodeGroupId(*f13iter.NodeGroupID)
+			}
+			if f13iter.ReplicaCount != nil {
+				f13elem.SetNewReplicaCount(*f13iter.ReplicaCount)
+			}
+			f13elemf2 := []*string{}
+			if f13iter.PrimaryAvailabilityZone != nil {
+				f13elemf2 = append(f13elemf2, &(*f13iter.PrimaryAvailabilityZone))
+			}
+			if f13iter.ReplicaAvailabilityZones != nil {
+				for _, f13elemf2iter := range f13iter.ReplicaAvailabilityZones {
+					var f13elemf2elem string
+					f13elemf2elem = *f13elemf2iter
+					f13elemf2 = append(f13elemf2, &f13elemf2elem)
+				}
+				f13elem.SetPreferredAvailabilityZones(f13elemf2)
+			}
+			f13 = append(f13, f13elem)
+		}
+		res.SetReplicaConfiguration(f13)
+	}
+
+	return res, nil
+}
+
+// new(DecreaseReplicaCount)RequestPayload returns an SDK-specific struct for the HTTP request
+// payload of the Create API call for the resource
+func (rm *resourceManager) newDecreaseReplicaCountRequestPayload(
+	r *resource,
+) (*svcsdk.DecreaseReplicaCountInput, error) {
+	res := &svcsdk.DecreaseReplicaCountInput{}
+
+	res.SetApplyImmediately(true)
+	if r.ko.Spec.ReplicationGroupID != nil {
+		res.SetReplicationGroupId(*r.ko.Spec.ReplicationGroupID)
+	}
+	if r.ko.Spec.NumNodeGroups != nil {
+		res.SetNewReplicaCount(*r.ko.Spec.ReplicasPerNodeGroup)
+	}
+
+	if r.ko.Spec.NodeGroupConfiguration != nil {
+		f13 := []*svcsdk.ConfigureShard{}
+		for _, f13iter := range r.ko.Spec.NodeGroupConfiguration {
+			f13elem := &svcsdk.ConfigureShard{}
+			if f13iter.NodeGroupID != nil {
+				f13elem.SetNodeGroupId(*f13iter.NodeGroupID)
+			}
+			if f13iter.ReplicaCount != nil {
+				f13elem.SetNewReplicaCount(*f13iter.ReplicaCount)
+			}
+			f13elemf2 := []*string{}
+			if f13iter.PrimaryAvailabilityZone != nil {
+				f13elemf2 = append(f13elemf2, &(*f13iter.PrimaryAvailabilityZone))
+			}
+			if f13iter.ReplicaAvailabilityZones != nil {
+				for _, f13elemf2iter := range f13iter.ReplicaAvailabilityZones {
+					var f13elemf2elem string
+					f13elemf2elem = *f13elemf2iter
+					f13elemf2 = append(f13elemf2, &f13elemf2elem)
+				}
+				f13elem.SetPreferredAvailabilityZones(f13elemf2)
+			}
+			f13 = append(f13, f13elem)
+		}
+		res.SetReplicaConfiguration(f13)
+	}
+
+	return res, nil
+}
+
+// This method copies the data from given replicationGroup by populating it into copy of supplied resource
+// and returns it.
+func provideUpdatedResource(
```

Comment:
:+1: I know it's a bit of a copy/paste pain right now, but we'll work together to auto-generate these kind of helper setter methods in a future PR :)

Reply:
👍